orbit/cmd/orbit/orbit.go

```diff
@@ -516,6 +518,36 @@ func main() {
 		}
 		g.Add(flagRunner.Execute, flagRunner.Interrupt)
 
+		// only setup extensions autoupdate if we have enabled updates
+		if !c.Bool("disable-updates") || c.Bool("dev-mode") {
+			const orbitExtensionUpdateInterval = 60 * time.Second
+			extRunner := update.NewExtensionConfigUpdateRunner(orbitClient, update.ExtensionUpdateOptions{
+				CheckInterval: orbitExtensionUpdateInterval,
+				RootDir:       c.String("root-dir"),
+			}, updateRunner)
+
+			if _, err := extRunner.DoExtensionConfigUpdate(); err != nil {
+				// just log, OK to continue since this will get retry
+				log.Info().Err(err).Msg("initial update to fetch extensions from /config API failed")
+			}
+
+			// call UpdateAction on the updateRunner after we have fetched extensions from Fleet
+			_, err := updateRunner.UpdateAction()
+			if err != nil {
+				// OK, initial call may fail, ok to continue
+				log.Debug().Err(err).Msg("extensions update action failed")
+			}
+
+			extensionAutoLoadFile := filepath.Join(c.String("root-dir"), "extensions.load")
+			stat, err := os.Stat(extensionAutoLoadFile)
+			// we only want to add the extensions_autoload flag to osquery, if the file exists and size > 0
+			if !errors.Is(err, os.ErrNotExist) && stat.Size() > 0 {
```

Comment:
Let's log the error if `err != nil` and `!errors.Is(err, os.ErrNotExist)`. The usual `switch` statement, something like:
```go
// We only want to add the extensions_autoload flag to osquery if the file exists and size > 0.
switch {
case err == nil:
  if stat.Size() > 0 {
    log.Debug().Msg("adding --extensions_autoload flag for file " + extensionAutoLoadFile)
    options = append(options, osquery.WithFlags([]string{"--extensions_autoload", extensionAutoLoadFile}))
  } else {
    log.Info(... empty extensions file ...)
  }
case errors.Is(err, os.ErrNotExist):
  // OK, nothing to do.
default:
  log.Error(...)
}
```

Reply:
Ah okay, any reason to prefer `switch/case` over `if` here? Maybe it's just me, but `if` reads a bit cleaner..? Happy to change it though, since that's the pattern we use often

Reply:
No preference, as long as we cover the mentioned case.

Reply:
updated